### PR TITLE
feat(plugin-chart-pivot-table): column, date and conditional formatting

### PR DESCRIPTION
**Pull request by kgabryje**
This PR introduces 4 new features and 1 cosmetic change to Pivot Table v2 chart.

1. Conditional formatting
    Feature described in https://github.com/apache/superset/pull/15651. This feature replaces heatmap and barchart table renderers. 

https://user-images.githubusercontent.com/15073128/125607141-c3b4e14d-d01d-4934-86fd-ecd499c04c1c.mov

2. Metrics d3 formats from dataset
    If there are custom d3 formats defined for saved metrics in dataset, Pivot table applies them to metric's values with higher priority than value formatting selected in "Customize" tab.

https://user-images.githubusercontent.com/15073128/125607278-c787d6b3-dad4-4d79-98ab-838c9b8147e5.mov

3. Date formatting
    Enabled applying d3 date formatting through a select component in "Customize" tab.

https://user-images.githubusercontent.com/15073128/125607318-9c657f4b-ac98-407b-aade-6d02855c4b6f.mov

4. Combine metrics
    Added a checkbox which allows using metrics as the lowest level group of columns or rows (before metrics were used only as a top level group). Added for feature parity with the old Pivot table chart. 

https://user-images.githubusercontent.com/15073128/125607380-cf939e03-098c-4f02-bd68-cff7143034a4.mov

5. Replaced collapse/expand icons from arrows to +/- icons from antd.

CC @junlincc @villebro @rusackas

### CODE REVIEW

**Comment by michael-s-molina** [comment]
@kgabryje Watching the video I noticed a problem with the validation message.

Screen Shot 2021-07-15 at 9 38 23 AM

Shouldn't be "This value should be `greater` than the `left` target value"?

Another thing, can you play with the layout to try to avoid the flicking effect that is happening when the validation message appears? Maybe increase the width of the fields, use shorter texts, reposition the fields, etc..

**Comment by kgabryje** [comment]
@michael-s-molina That's what you get when you copy paste a message and forget to modify it 🤦  Anyway, the things that you point out are defined in the main superset repo - let's fix that in a separate PR.

**Review by michael-s-molina — APPROVED** [review]
LGTM assuming that the tests will be done in `superset` and the adjustments in separate PRs.

File: plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts [hunk 1]
```diff
@@ -16,9 +16,28 @@
...
+} from '@superset-ui/core';
+import { getColorFormatters } from '@superset-ui/chart-controls';
+import { DateFormatter } from '../types';
 
-export default function transformProps(chartProps: ChartProps) {
+const { DATABASE_DATETIME } = TimeFormats;
+const TIME_COLUMN = '__timestamp';
+
+function isNumeric(key: string, data: DataRecord[] = []) {
+  return data.every(x => x[key] === null || x[key] === undefined || typeof x[key] === 'number');
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts]: done 👍

**Comment by Prateekarora1998** [comment]
I am currently using conditional formatting in pivot table in the superset. Can anyone please tell me how to add colours in the drop down menu (color scheme menu)?

**Comment by villebro** [comment]
> I am currently using conditional formatting in pivot table in the superset. Can anyone please tell me how to add colours in the drop down menu (color scheme menu)?

@Prateekarora1998 currently adding custom values isn't possible. However, we're planning on adding full color gradient support later that uses the linear color scheme control (see screenshot below), but I'm unsure when we'll be able to get to it. If you're interested in contributing to the feature please let us know!

image

**Pull request merged**